### PR TITLE
Payment Methods: Add REST API integration for Add Credit Card page

### PR DESCRIPTION
**Pull request by gziolo**
Final part of the Add Credit Card page implementation for Connect for WooCommerce (see #6971 for more details).

This PR adds new endpoint integration that allows to store credit card without any purchase assigned. Props to @prettyboymp for backend work.

### Testing.
1. Open new Add Credit Card page (http://calypso.localhost:3000/payment-methods/add-credit-card or https://calypso.live/payment-methods/add-credit-card).
2. Provide invalid credit card data that will pass form validation but trigger an error (date in the past should do the trick). You should see error message:
screen shot 2016-08-23 at 12 15 46
3. Provide credit card data that will succeed. You should be redirected to the Billing History page. Newly added credit card should be listed on the page. There should be also a success message displayed.
screen shot 2016-08-23 at 14 02 46
4. Try to use newly added card in the regular purchase flow! Enjoy :)

Test live: https://calypso.live/?branch=update/add-credit-card-api

### CODE REVIEW

**Comment by breezyskies** [comment]
> I'm not quite sure if TOS text fits well in this context. This page allows to add credit card without any linked purchase. So use might use it to purchase one time purchase, too.

This is a question for @jennyzhu. 

One of the UX issues I see with this way of handling the credit card on file is that the "general usage" doesn't make this type of thing clear. Is there a place where they _would_ agree to a TOS, if not here?

**Comment by prettyboymp** [comment]
LGTM! 👍

**Comment by nabsul** [comment]
When I enter a valid credit card number but with invalid CVV, I get a very technical 404 error message:

image

**Comment by nabsul** [comment]
Hmmm... actually, I get the same error even for a valid credit card.

**Comment by janm6k** [comment]
@nabsul Are you sure you are testing correct branch with recent changes? I got the same error as you when I tried master, but on `update/add-credit-card-api` it worked correctly.

**Comment by allendav** [comment]
@breezyskies @jennyzhu @gziolo @ranh fwiw, connect (the requestors of this feature) will be prompting users with a TOS. Can't speak for others of course.

**Comment by nabsul** [comment]
@janm6k I haven't tried to set up my own local Calypso environment. I was trying https://calypso.live/payment-methods/add-credit-card

Is this expected in that case?

**Comment by nabsul** [comment]
Oh! I missed the very last line of the instructions! Let me try again.

**Comment by jennyzhu** [comment]
@gziolo @breezyskies 

> I'm not quite sure if TOS text fits well in this context. This page allows to add credit card without any linked purchase. So use might use it to purchase one time purchase, too.

Genericized the language a bit. Can we use this?

By saving a credit card, you agree to our [Terms of Service](https://en.wordpress.com/tos/), and if you use it to pay for a subscription or plan, you authorize your credit card to be charged on a recurring basis until you cancel, which you can do at any time. You understand [how your subscription works](https://support.wordpress.com/auto-renewal) and [how to cancel](https://support.wordpress.com/manage-purchases).

**Comment by gziolo** [comment]
@jennyzhu Sounds good. Thanks!

@breezyskies TOS text is updated now:
screen shot 2016-08-24 at 09 23 40
Does it look good for you?

@nabsul Were you able to test it with success?

When should we enable this feature for all users?

**Comment by breezyskies** [comment]
Design 👍

**Comment by nabsul** [comment]
When I load the page on calypso.live, I get the following message in the console. Is this expected?
image

**Comment by nabsul** [comment]
Also, I seem to be able to add credit cards with invalid expiration, CVV and postcode.

**Comment by gziolo** [comment]
> When I load the page on calypso.live, I get the following message in the console. Is this expected?

Warning issue was fixed in another PR today: https://github.com/Automattic/wp-calypso/pull/7632. I haven't rebased this branch yet.

> Also, I seem to be able to add credit cards with invalid expiration, CVV and postcode.

@prettyboymp do we validate cards on the backend side?

**Comment by prettyboymp** [comment]
@gziolo There isn't currently any input validation on the endpoint because it only receives the token.  However, the token gets validated against payment service before adding the card to the account.  If the user is sandboxed and set to use the billing sandbox as well, the payment service will accept invalid cards, though I don't believe it should be for invalid dates.

**Comment by nabsul** [comment]
Sounds good. Those are all my concerns. Thanks for pushing this forward!
